client/allocrunner/taskrunner/getter/util_linux.go

```diff
@@ -42,12 +43,35 @@ func credentials() (uint32, uint32) {
 	return userUID, userGID
 }
 
+// findHomeDir returns the home directory as provided by os.UserHomeDir. In case
+// os.UserHomeDir returns an error, we return /root if the current process is being
+// run by root, or /dev/null otherwise.
+func findHomeDir() string {
+	// When running as a systemd unit the process may not have the $HOME
+	// environment variable set, and os.UserHomeDir will return an error.
+
+	// if home is set, just use that
+	if home, err := os.UserHomeDir(); err == nil {
+		return home
+	}
+
+	// if we are the root user, use "/root"
+	if unix.Getuid() == 0 {
+		return "/root"
+	}
```

Comment:
We have `mitchell/go-homedir` in our dependencies already, which checks `getent` to get the homedir out of `/etc/passwd` if `$HOME` isn't set: https://github.com/mitchellh/go-homedir/blob/main/homedir.go#L114 Should we use that to handle weird edge cases like `$HOME` being unset for non-root users?

Reply:
Good call! Swapped out UserHomeDir for go-homedir; this makes the tests a bit more interesting to setup but seems worth it.